test/functional/kernel/hyperplane/test-kernel-hyperplane-2D.cpp.in

```diff
@@ -0,0 +1,212 @@
+
+// for data types
+#include "RAJA_test-reduce-types.hpp"
+#include "RAJA_test-forall-data.hpp"
+
+//
+// Header for tests in ./tests directory
+//
+// Note: CMake adds ./tests as an include dir for these tests.
+//
+#include "test-kernel-hyperplane-@TEST_TYPE@.hpp"
+
+
+//
+// Exec pols for kernel tile tests
+//
+
+using SequentialKernelHyperplaneExecPols =
+  camp::list<
+
+    RAJA::KernelPolicy<
+      RAJA::statement::For<0, RAJA::seq_exec,
+        RAJA::statement::Hyperplane<2, RAJA::seq_exec, RAJA::ArgList<1>, RAJA::seq_exec,
+          RAJA::statement::Lambda<0>
+        >
+      >
+    >,
+
+    RAJA::KernelPolicy<
+      RAJA::statement::For<0, RAJA::seq_exec,
+        RAJA::statement::Hyperplane<1, RAJA::seq_exec, RAJA::ArgList<2>, RAJA::seq_exec,
+          RAJA::statement::Lambda<0>
+        >
+      >
+    >
+
+  >;
+
+#if defined(RAJA_ENABLE_OPENMP)
+
+using OpenMPKernelHyperplaneExecPols =
+  camp::list< 
+
+    RAJA::KernelPolicy<
+      RAJA::statement::For<0, RAJA::omp_parallel_for_exec,
+        RAJA::statement::Hyperplane<1, RAJA::seq_exec, RAJA::ArgList<2>, RAJA::seq_exec,
+          RAJA::statement::Lambda<0>
+        >
+      >
+    >,
+
+    RAJA::KernelPolicy<
+      RAJA::statement::For<0, RAJA::omp_parallel_for_exec,
+        RAJA::statement::Hyperplane<2, RAJA::seq_exec, RAJA::ArgList<1>, RAJA::seq_exec,
+          RAJA::statement::Lambda<0>
+        >
+      >
+    >
+  >;
+
+#endif  // RAJA_ENABLE_OPENMP
+
+#if defined(RAJA_ENABLE_TARGET_OPENMP)
+
+using OpenMPTargetKernelHyperplaneExecPols =
+  camp::list< 
+
+    RAJA::KernelPolicy<
+      RAJA::statement::For<0, RAJA::omp_target_parallel_for_exec<16>,
+        RAJA::statement::Hyperplane<1, RAJA::seq_exec, RAJA::ArgList<2>, RAJA::seq_exec,
+          RAJA::statement::Lambda<0>
+        >
+      >
+    >,
+
+    RAJA::KernelPolicy<
+      RAJA::statement::For<0, RAJA::omp_target_parallel_for_exec<16>,
+        RAJA::statement::Hyperplane<2, RAJA::seq_exec, RAJA::ArgList<1>, RAJA::seq_exec,
+          RAJA::statement::Lambda<0>
+        >
+      >
+    >
+  >;
+
+#endif  // RAJA_ENABLE_TARGET_OPENMP
+
+#if defined(RAJA_ENABLE_TBB)
+
+using TBBKernelHyperplaneExecPols =
+  camp::list< 
+
+    RAJA::KernelPolicy<
+      RAJA::statement::For<0, RAJA::tbb_for_exec,
+        RAJA::statement::Hyperplane<1, RAJA::seq_exec, RAJA::ArgList<2>, RAJA::seq_exec,
+          RAJA::statement::Lambda<0>
+        >
+      >
+    >,
+
+    RAJA::KernelPolicy<
+      RAJA::statement::For<0, RAJA::tbb_for_exec,
+        RAJA::statement::Hyperplane<2, RAJA::seq_exec, RAJA::ArgList<1>, RAJA::seq_exec,
+          RAJA::statement::Lambda<0>
+        >
+      >
+    >
+  >;
+
+#endif  // RAJA_ENABLE_TBB
+
+#if defined(RAJA_ENABLE_CUDA)
+
+using CudaKernelHyperplaneExecPols =
+  camp::list< 
+
+    RAJA::KernelPolicy<
+      RAJA::statement::CudaKernel<
+        RAJA::statement::For<0, RAJA::cuda_block_x_loop,
+          RAJA::statement::For<1, RAJA::cuda_thread_x_direct,
```

Comment:
Is it correct to have nested CUDA "loops" using the same thread dimension, in this case 'x'?

Reply:
Yes, that is valid code.

Reply:
The thread and block indices are independent of each other.